lib/api.js

```diff
@@ -17,22 +28,43 @@ module.exports = (self, options) => {
         .toArray();
 
       return dialogs;
-    } catch (error) {
-      self.apos.utils.warn(error);
-
+    } catch (err) {
+      self.apos.utils.error(err);
       return { status: 'error', message: 'Something went wrong' };
     }
   };
 
   self.findOne = async req => {
     try {
-      const dialog = await self.apos.modules['apostrophe-dialog-box']
-        .find(req, { _id: req.params.id })
-        .toObject();
+      let dialog;
+      if (req.session.workflowMode && req.session.workflowMode === 'live') {
+        // rich text dialog links will always contain the draft id,
+        // transpose this to live if necessary
+
+        // we could still have a good join, give it a second
+        dialog = await self.apos.modules['apostrophe-dialog-box']
+          .find(req, { _id: req.params.id })
+          .toObject();
 
-      return dialog;
-    } catch (error) {
-      return null;
+        if (!dialog) {
+          // at this point we likely need to go look for the live version
+          const get = Promise.promisify(self.getDraftAndLiveWrapper);
```

Comment:
How did this problem get started? How is the id of the dialog box being stored? If it is a joinByOne then committing should be translating it to the id of the live version of the same dialog.